client/src/domain/event/EventHero.tsx

```diff
@@ -0,0 +1,285 @@
+import classNames from "classnames";
+import {
+  isSameDay,
+  isSameMonth,
+  isSameYear,
+  isThisWeek,
+  isToday
+} from "date-fns";
+import capitalize from "lodash/capitalize";
+import React from "react";
+import { useTranslation } from "react-i18next";
+import { RouteComponentProps, withRouter } from "react-router";
+
+import Button from "../../common/components/button/Button";
+import Keyword from "../../common/components/keyword/Keyword";
+import { EventDetailsQuery } from "../../generated/graphql";
+import AngleRightIcon from "../../icons/AngleRightIcon";
+import ArrowLeftIcon from "../../icons/ArrowLeftIcon";
+import CalendarIcon from "../../icons/CalendarIcon";
+import LocationIcon from "../../icons/LocationIcon";
+import ShareIcon from "../../icons/ShareIcon";
+import TicketIcon from "../../icons/TicketIcon";
+import { formatDate } from "../../util/dateUtils";
+import getLocale from "../../util/getLocale";
+import getLocalisedString from "../../util/getLocalisedString";
+import Container from "../app/layout/Container";
+import styles from "./eventHero.module.scss";
+
+interface Props extends RouteComponentProps {
+  eventData: EventDetailsQuery;
+}
+
+const EventHero: React.FC<Props> = ({
+  eventData,
+  history: { push },
+  location: { search }
+}) => {
+  const { t } = useTranslation();
+  const locale = getLocale();
+
+  const getLocationStr = () => {
+    const location = eventData.linkedEventsEventDetails.location;
+    const addressLocality = getLocalisedString(
+      (location && location.addressLocality) || {},
+      locale
+    );
+    const locationName = getLocalisedString(
+      (location && location.name) || {},
+      locale
+    );
+    const streetAddress = getLocalisedString(
+      (location && location.streetAddress) || {},
+      locale
+    );
+
+    return [locationName, streetAddress, addressLocality].join(", ");
+  };
+
+  const getDateStr = (start: string, end: string | null | undefined) => {
```

Comment:
Would it make sense to have this function as an utility function in some separate file?